synapse/storage/databases/main/events.py

```diff
@@ -1183,6 +1183,12 @@ def _update_current_state_txn(
                 txn, room_id, members_changed
             )
 
+            # Check if any of the remote membership changes requires us to
+            # unsubscribe from their device lists.
+            self.store.handle_potentially_left_users_txn(
+                txn, {m for m in members_changed if not self.hs.is_mine_id(m)}
+            )
```

Comment:
Oh, we no longer check if the old memberships were joins, which makes this slightly more expensive than at present.